### PR TITLE
mongodb_user: Add new API for user management

### PR DESCRIPTION
**Pull request by carlochess**
##### SUMMARY
add_user is deprecated and will be removed in PyMongo 4.0. Starting
with MongoDB 2.6 user management is handled with four database
commands, createUser, usersInfo, updateUser, and dropUser.

Please see: http://api.mongodb.com/python/current/api/pymongo/database.html

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
mongodb_user

##### ADDITIONAL INFORMATION
No additional information.

### CODE REVIEW

**Comment by gundalow** [comment]
bot_status
Not sure why maintainers were not pinged

@elliotttf @Lujeni @nerzhul I believe you have worked on the mongo modules. This patch seems straightforward and correct, thanks @carlochess, though I'd like someone else to review before it gets merged.

Once merged into `devel` this would be good to get into `stable-2.7` by following https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs

**Comment by gundalow** [comment]
@elliotttf @Lujeni @nerzhul could you please take a look at this patch, thanks in advance

**Comment by dagwieers** [comment]
We need MongoDB integration tests before we can change modules and behavior. See https://github.com/ansible/community/wiki/MongoDB to collaborate on integration tests.

File: lib/ansible/modules/database/mongodb/mongodb_user.py [hunk 1]
```diff
@@ -268,7 +271,10 @@ def user_remove(module, client, db_name, user):
         if module.check_mode:
             module.exit_json(changed=True, user=user)
         db = client[db_name]
-        db.remove_user(user)
+        if LooseVersion(PyMongoVersion) < LooseVersion('4.0'):
```
Comment (Lujeni) [inline lib/ansible/modules/database/mongodb/mongodb_user.py:274]:
```
if LooseVersion(PyMongoVersion) < LooseVersion('4.0'):
    db.remove_user(user)
db.command("dropUser", user)
```
You can avoid if/else in this case, improve code visibility

File: lib/ansible/modules/database/mongodb/mongodb_user.py [hunk 2]
```diff
@@ -256,10 +256,13 @@ def user_add(module, client, db_name, user, password, roles):
     # without reproducing a lot of the logic in database.py of pymongo
     db = client[db_name]
 
-    if roles is None:
-        db.add_user(user, password, False)
+    if LooseVersion(PyMongoVersion) < LooseVersion('4.0'):
+        if roles is None:
+            db.add_user(user, password, False)
+        else:
```
Comment (Lujeni) [inline lib/ansible/modules/database/mongodb/mongodb_user.py:262]: You can avoid if/else in this case also

File: lib/ansible/modules/database/mongodb/mongodb_user.py (hunk 1 above)
Comment (carlochess) [reply lib/ansible/modules/database/mongodb/mongodb_user.py:274]: Let me check if this would work for Mongo versions < 4.0

**Pull request closed without merging**